objective-c/route_guide/README.md

```diff
@@ -0,0 +1,280 @@
+
+With gRPC you can define your service once in a .proto file and implement clients and servers in any of gRPC's supported languages, which in turn can be run in environments ranging from servers inside Google to your own tablet - all the complexity of communication between different languages and environments is handled for you by gRPC. You also get all the advantages of working with protocol buffers, including efficient serialization, a simple IDL, and easy interface updating.
+
+gRPC and proto3 are specially suited for mobile clients: gRPC is implemented on top of HTTP/2, which results in network bandwidth savings over using HTTP/1.1. Serialization and parsing of the proto binary format is more efficient than the equivalent JSON, resulting in CPU and battery savings. And proto3 uses a runtime that has been optimized over the years at Google to keep code size to a minimum. The latter is important in Objective-C, because the ability of the compiler to strip unused code is limited by the dynamic nature of the language.
+
+
+<a name="setup"></a>
+## Example code and setup
+
+The example code for our tutorial is in [grpc/grpc-common/objective-c/route_guide](https://github.com/grpc/grpc-common/tree/master/objective-c/route_guide). To download the example, clone the `grpc-common` repository by running the following command:
+```shell
+$ git clone https://github.com/grpc/grpc-common.git
+```
+
+Then change your current directory to `grpc-common/objective-c/route_guide`:
+```shell
+$ cd grpc-common/objective-c/route_guide
+```
+
+Our example is a simple route mapping application that lets clients get information about features on their route, create a summary of their route, and exchange route information such as traffic updates with the server and other clients.
+
+You also should have [Cocoapods](https://cocoapods.org/#install) installed, as well as the relevant tools to generate the client library code (and a server in another language, for testing). You can obtain the latter by following [these setup instructions](https://github.com/grpc/homebrew-grpc).
+
+
+<a name="try"></a>
+## Try it out!
+
+To try the sample app, we need a gRPC server running locally. Let's compile and run, for example, the C++ server in this repository:
+
+```shell
+$ pushd ../../cpp/route_guide
+$ make
+$ ./route_guide_server &
+$ popd
+```
+
+Now have Cocoapods generate and install the client library for our .proto files:
+
+```shell
+$ pod install
+```
+
+(This might have to compile OpenSSL, which takes around 15 minutes if Cocoapods doesn't have it yet on your computer's cache).
+
+Finally, open the XCode workspace created by Cocoapods, and run the app. You can check the calling code in `ViewControllers.m` and see the results in XCode's log console.
+
+The next sections guide you step-by-step through the creation of this proto service, the client library generated from it, and the app that uses that library.
+
+
+<a name="proto"></a>
+## Defining the service
+
```

Comment:
True, but OTOH at no point are we telling her in this tutorial how to create a server - you're unlikely to define a service and then just write clients for it even though it doesn't have any servers. Unless it's a reasonably likely scenario that an Objective C user would define a service, get someone else to implement the server in another language, then create a client app?

That's why I suggested rephrasing it as "let's look at how this service is defined" rather than "first you need to define this service", as the service is clearly already defined and implemented on the server side (and indeed running) in our scenario. It still provides enough info for the user to create their own service if they so wish.